src/utils/handleRequestWith.ts

```diff
@@ -64,7 +64,18 @@ export const handleRequestWith = (
       req.body = parseBody(req.body, req.headers)
 
       // Set document cookies on the request.
-      req.cookies = getRequestCookies(req)
+      store.hydrate()
+      req.cookies = Object.fromEntries(
```

Comment:
Can we move this logic into `getRequestCookies` function? 
`req.cookies` and `req.headers.set` blocks are quite verbose, and are inseparable, so I'd vote for a standalone function that we could unit test.

Reply:
I created a `setRequestCookies()` and a `readResponseCookies()` function. I think that also resolves the comment below.